### PR TITLE
Hods 174 - Adding CT endpoints and domain models

### PR DESCRIPTION
**Pull request by matthewharris1**

### CODE REVIEW

File: app/uk/gov/hmrc/organisationsdetailsapi/controllers/CorporationTaxController.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+          auditHelper.auditCorporationTaxApiResponse(correlationId.toString, matchId.toString,
+            authScopes.mkString(","), request, selfLink.toString, Some(Json.toJson(corporationTax)))
+
+          Ok(response)
+        }
+      } recover recoveryWithAudit(maybeCorrelationId(request), matchId.toString, "/organisations/details/corporation-tax")
+  }
+  override val environment: String = PRODUCTION
+  // TODO: add me to the routes files
```
Comment (Luka-Abey) [inline app/uk/gov/hmrc/organisationsdetailsapi/controllers/CorporationTaxController.scala:62]: as far as I can see, this has been done? If so, please remove comment :)
```suggestion
```

File: test/unit/uk/gov/hmrc/organisationsdetailsapi/audit/AuditHelperSpec.scala [hunk 2]
```diff
@@ -139,4 +139,7 @@ class AuditHelperSpec  extends AsyncWordSpec with Matchers with MockitoSugar {
     capturedEvent.returnLinks shouldEqual ifUrl
     capturedEvent.applicationId shouldBe applicationId
   }
+
+  //TODO: Add test for auditing Corporation tax response
```
Comment (Luka-Abey) [inline test/unit/uk/gov/hmrc/organisationsdetailsapi/audit/AuditHelperSpec.scala:143]: also here - looks like its been added to corpTaxControllerSpec
```suggestion
```

**Review by Luka-Abey — CHANGES_REQUESTED** [review]
Just a couple comments about comments

File: test/component/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnectorSpec.scala [hunk 3]
```diff
@@ -67,6 +67,8 @@ class IfConnectorSpec
     .bindings(bindModules: _*)
     .configure(
       "cache.enabled"  -> false,
+      "auditing.enabled"                                       -> false,
+      "auditing.traceRequests"                                 -> false,
       "microservice.services.integration-framework.host" -> "localhost",
```
Comment (ajchilts) [inline test/component/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnectorSpec.scala:72]: In order to comply with the latest auth upgrade please change the host to `127.0.0.1`. This means that it routes through auth correctly as localhost will bypass.

File: test/component/uk/gov/hmrc/organisationsdetailsapi/stubs/BaseSpec.scala [hunk 4]
```diff
@@ -28,27 +33,62 @@ import play.mvc.Http.MimeTypes.JSON
...
 
 trait BaseSpec
-  extends AnyFeatureSpec with BeforeAndAfterAll with BeforeAndAfterEach with GuiceOneServerPerSuite
+  extends AnyFeatureSpec with BeforeAndAfterAll with BeforeAndAfterEach with Matchers with GuiceOneServerPerSuite
     with GivenWhenThen {
 
   implicit override lazy val app: Application = GuiceApplicationBuilder()
     .configure(
+      "mongodb.uri"                             -> "mongodb://127.0.0.1:27017/organisations-details-api",
+      "microservice.services.integration-framework.host" -> "localhost",
```
Comment (ajchilts) [inline test/component/uk/gov/hmrc/organisationsdetailsapi/stubs/BaseSpec.scala:42]: In order to comply with the latest auth upgrade please change the host to 127.0.0.1. This means that it routes through auth correctly as localhost will bypass.

**Review by ajchilts — CHANGES_REQUESTED** [review]

**Review by owen-opencast — CHANGES_REQUESTED** [review]

**Review by owen-opencast — CHANGES_REQUESTED** [review]
Looks good, just spotted we need the default port specifying in build.sbt (see orgs matching)

**Review by owen-opencast — APPROVED** [review]

**Review by ajchilts — APPROVED** [review]
Looking good now 👍

**Pull request merged**